packages/bvaughn-architecture-demo/components/console/MessagesList.tsx

```diff
@@ -86,6 +88,8 @@ function MessagesList({ forwardedRef }: { forwardedRef: ForwardedRef<HTMLElement
     const loggableExecutionPoint = getLoggableExecutionPoint(loggable);
     const isLoaded =
       loadedRegions !== null && isPointInRegions(loggableExecutionPoint, loadedRegions.loaded);
+    const includesFirstConsoleNag = currentUserInfo?.nags?.includes(Nag.FIRST_CONSOLE_NAVIGATE);
+    const initialIsHovered = Boolean(!includesFirstConsoleNag && index === 0);
```

Comment:
Hmm. Item renderers already have an `index` prop, so a renderer can already know if it's the first item in the list. (Admittedly this wouldn't account for the case where you filter items in the list such that the actual first one doesn't get shown– but I don't think this PR is account for that case either? and I don't think we need to, to be honest.)

I wonder if renderers should check this condition directly?

```js
const showInitialConsoleHoverPrompt = currentUserInfo.onboarding.showInitialConsoleHoverPrompt;
const [isHovered, setIsHovered] = useState(showInitialConsoleHoverPrompt);
```

I don't think _all_ renderer types would need this check either. Maybe just `LogPointRenderer`?

If it's the first time they're using Replay, they won't have added a terminal expression or a log point yet– so those renderers seem unnecessary.

Event logs or uncaught exception logs renderers _could_ check this, although those require the user to actively ask to see the data– so maybe they wouldn't even need to? On the fence about it.

Reply:
Ah, yeah I think this was overkill adding to each renderer! I cleaned this up to only change the initial hover state of the `LogPointRenderer`.

Thinking through this more I think it will need to be a little smarter since it's checking purely index right now, but should be whatever the _first_ log point is. Assuming that shouldn't be hard to calculate though.